api/lib/opentelemetry/context.rb

```diff
@@ -29,23 +30,45 @@ def current
         Thread.current[KEY] ||= ROOT
       end
 
-      # Sets the current context
-      #
-      # @param [Context] ctx The context to be made active
+      # @api private
       def current=(ctx)
         Thread.current[KEY] = ctx
       end
 
+      # Returns a token to be used with the matching call to detach
+      #
+      # @param [Context] context The new context
+      # @return [Integer] A token to be used when detaching
+      def attach(context)
+        prev = current
+        self.current = context
+        stack.push(prev)
+        stack.size
+      end
+
+      # Restores the previous context, if a token is supplied it will
+      # be used to check if the call to detach is balanced with
+      # the original attach call. A warning is logged if the
+      # calls are unbalanced.
+      #
+      # @param [Integer] token The token provided by the matching call to attach
```

Comment:
```suggestion
      # @param [Integer] token The token provided by the matching call to attach
      # @return [Boolean] True if the calls matched, false otherwise
```

Reply:
I think it's useful to return this indication, even if it isn't used. The spec suggests:
> The API MAY return a value used to check whether the operation was successful or not.